tests/kernel/run_ltp.pm

```diff
@@ -0,0 +1,303 @@
+        else {
+            $ignored_lines++;
+        }
+    }
+
+    return ($ignored_lines, $tconf, $tfail);
+}
+
+sub parse_openposix_log {
+    my ($self, $test_log, $fin_msg, $fh) = @_;
+    my ($tconf, $tfail) = (0, 0);
+
+    $test_log =~ m/$fin_msg(\d+)/;
+    print $fh 'Test process returned ';
+    if ($1 eq '0') {
+        print $fh 'PASSED';
+    }
+    elsif ($1 eq '1') {
+        print $fh 'FAILED';
+        $tfail = 1;
+    }
+    elsif ($1 eq '2') {
+        print $fh 'UNRESOLVED';
+        $tconf = 1;
+    }
+    elsif ($1 eq '4') {
+        print $fh 'UNSUPPORTED';
+        $tconf = 1;
+    }
+    elsif ($1 eq '5') {
+        print $fh 'UNTESTED';
+        $tconf = 1;
+    }
+    else {
+        print $fh 'unknown';
+        $tfail = 1;
+    }
+    say $fh " ($1) exit code.";
+    return (0, $tconf, $tfail);
+}
+
+sub record_ltp_result {
+    my ($self, $name, $test_log, $fin_msg, $duration, $is_posix) = @_;
+    my ($details, $fh) = $self->start_result($name, $name);
+    my $ignored_lines = 0;
+    my ($tconf, $tfail) = (0, 0);
+
+    unless (defined $test_log) {
+        print $fh 'This test took too long to complete!';
+        $details->{result} = 'fail';
+        close $fh;
+        push @{$self->{details}}, $details;
+        die "Can't continue; timed out waiting for LTP test case which may still be running or the OS may have crashed!";
+    }
+
+    if ($is_posix) {
+        ($ignored_lines, $tconf, $tfail) = $self->parse_openposix_log($test_log, $fin_msg, $fh);
+    }
+    else {
+        ($ignored_lines, $tconf, $tfail) = $self->parse_ltp_log($test_log, $fin_msg, $fh);
+    }
+
+    if ($tfail) {
+        $details->{result} = 'fail';
+        $self->{result}    = 'fail';
+    }
+    elsif ($tconf) {
+        $details->{result} = 'unk';
+    }
+
+    say $fh "Test took approximately $duration seconds";
+
+    if ($ignored_lines > 0) {
+        print $fh "Some test output could not be parsed: $ignored_lines lines were ignored.";
+    }
+
+    $self->commit_result($details, $fh);
+}
+
+sub run {
+    my ($self) = @_;
+    my $cmd_file = get_var 'LTP_COMMAND_FILE';
+    die 'Need LTP_COMMAND_FILE to know which tests to run' unless $cmd_file;
+    my $cmd_pattern = get_var 'LTP_COMMAND_PATTERN';
+    my $timeout     = get_var 'LTP_TIMEOUT' || 300;
+    my $is_posix    = $cmd_file =~ m/^\s*openposix\s*$/i;
+
+    my @tests;
+    if ($is_posix) {
+        my $cmd_file_text = script_output('cat ~/openposix_test_list.txt');
+        @tests = $self->parse_openposix_runfile($cmd_file_text, $cmd_pattern);
+    }
+    else {
+        my $cmd_file_text = script_output("cat /opt/ltp/runtest/$cmd_file");
+        @tests = $self->parse_runfile($cmd_file_text, $cmd_file, $cmd_pattern);
+    }
+
+    for my $test (@tests) {
+        my $fin_msg    = "### TEST $test->{name} COMPLETE >>> ";
+        my $cmd_text   = qq($test->{command}; echo "$fin_msg\$?");
+        my $start_time = gettimeofday;
+        if (is_serial_terminal) {
```

Comment:
Yep, if it was running on the SUT it would likely report bogus values for a few tests. But hey the `clock_gettime()` with `CLOCK_MONOTONIC` should be part of the `Time:HiRes` anyway, so switching to it should be just painless.